pkg/dataobj/logs_reader_test.go

```diff
@@ -184,3 +186,68 @@ func readAllRecords(ctx context.Context, r *dataobj.LogsReader) ([]dataobj.Recor
 		clear(buf)
 	}
 }
+
+// BenchmarkLogsReader benchmarks the performance of reading 10k log lines with structured metadata from a data object.
+func BenchmarkLogsReader(b *testing.B) {
+	// Build the object
+	opts := dataobj.BuilderConfig{
+		TargetObjectSize:  16 * 1024 * 1024,
+		TargetSectionSize: 4 * 1024 * 1024,
+		BufferSize:        16 * 1024 * 1024,
+		TargetPageSize:    2 * 1024 * 1024,
+	}
+
+	builder, err := dataobj.NewBuilder(opts)
+	require.NoError(b, err)
+
+	for i := 0; i < 10000; i++ {
+		err := builder.Append(push.Stream{
+			Labels: "{cluster=\"test\",app=\"bar\"}",
+			Entries: []push.Entry{
+				{
+					Timestamp: time.Now().Add(time.Duration(i) * time.Second),
+					Line:      "hello world " + strconv.Itoa(i),
+					StructuredMetadata: push.LabelsAdapter{
+						{Name: "trace_id", Value: "123"},
+						{Name: "pod", Value: "pod-abcd"},
+					},
+				},
+			},
+		})
+		require.NoError(b, err)
+	}
+
+	var buf bytes.Buffer
+	_, err = builder.Flush(&buf)
+	require.NoError(b, err)
+
+	obj := dataobj.FromReaderAt(bytes.NewReader(buf.Bytes()), int64(len(buf.Bytes())))
+
+	md, err := obj.Metadata(context.Background())
+	require.NoError(b, err)
+	require.Equal(b, 1, md.LogsSections)
+
+	r := dataobj.NewLogsReader(obj, 0)
+
+	var (
+		recs = make([]dataobj.Record, 128)
+		ctx  = context.Background()
+	)
+	b.ResetTimer()
+	b.ReportAllocs()
```

Comment:
I've seen people add this a few times, I'm curious if there's any benefits to calling `b.ReportAllocs()` (which forces allocs to always be benchmarked) vs letting the tester decide by passing `-benchmem` to go test (which turns on benchmarking allocs for anything that hasn't called ReportAllocs)?

Reply:
🤷 It's not really a huge difference, so I'm fine with it either way